### PR TITLE
Init DynamoDBQuery block

### PR DESCRIPTION
**Pull request by hansmosh**

### CODE REVIEW

File: tests/test_dynamo_db_query_block.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+from unittest.mock import MagicMock, patch
+from nio.common.signal.base import Signal
+from nio.util.support.block_test_case import NIOBlockTestCase
+from ..dynamo_db_query_block import DynamoDBQuery
+from ..dynamo_db_base_block import DynamoDBBase
+
+
+class SaveCounterDynamoDB(DynamoDBQuery):
```
Comment (mattdodge) [inline tests/test_dynamo_db_query_block.py]: This used anywhere?

File: dynamo_db_query_block.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        output = []
+        query_dict = {}
+        for query_filter in self.query_filters:
+            # evaluate query filter expression
+            try:
+                key = query_filter.key(signal)
+                value = query_filter.value(signal)
+                query_dict[key] = value
+            except:
+                self._logger.debug('Failed to evaluate query filter key/value')
```
Comment (mattdodge) [inline dynamo_db_query_block.py]: Probably should be a warning level at least. Since we re-raise anyway we might not even need the try/except here

**Comment by mattdodge** [comment]
Looks good, couple of comments but almost there

File: tests/test_dynamo_db_query_block.py (hunk 1 above)
Comment (hansmosh) [inline tests/test_dynamo_db_query_block.py]: Good catch. I copied it from the other test but ended up not using it.

File: dynamo_db_query_block.py (hunk 2 above)
Comment (hansmosh) [inline dynamo_db_query_block.py]: Yeah. The idea I'm going for is that execute_signals_query would always log to exception, but pieces of code deeper down could still catch and log to other levels to provide more detail. The benefit of that would be two provide more detail about the error if you're logging to debug.

**Pull request merged**